src/GeneratorV2/Process/Overloaders/VectorOverloader.cs

```diff
@@ -0,0 +1,188 @@
+                    int length = -1;
+                    if (parameter.Length is Constant constant)
+                    {
+                        length = constant.Value;
+                    }
+                    else if (parameter.Length is BinaryOperation binary)
+                    {
+                        // If we have a binary operation length we need to set the count parameter to 1.
+                        Expression other;
+                        if (binary.Left is Constant left)
+                        {
+                            length = left.Value;
+                            other = binary.Right;
+                        }
+                        else if (binary.Right is Constant right)
+                        {
+                            length = right.Value;
+                            other = binary.Left;
+                        }
+                        else
+                        {
+                            newOverloads = null;
+                            return false;
+                        }
+
+                        if (other is ParameterReference reference)
+                        {
+                            Parameter? countParameter = newParameters.Find(p => p.Name == reference.ParameterName);
+                            if (countParameter == null)
+                                throw new Exception($"{reference.ParameterName} is not a valid reference on {name}");
+                            countParameters.Add(countParameter);
+                        }
+                    }
+
+                    if (length < 2 || length > 4)
+                    {
+                        continue;
+                    }
+
+                    // Save the new parameter.
+                    nameTable.Rename(parameter, parameter.Name + "_ptr");
+                    string vectorType = "Vector" + length + (typeChar == 'f' ? "" : typeChar);
+                    newParameters[i] = parameter with
+                    {
+                        Type = new CSRef(csType.Constant ? CSRef.Type.In : CSRef.Type.Ref, new CSType(vectorType, csType.Constant)),
+                        Length = null
+                    };
+                    foundVector = true;
+                    pointerParameters.Add(parameter);
+                    vectorParameters.Add(newParameters[i]);
+                }
+            }
+
+            if (foundVector)
+            {
+                // Remove the count parameters from the new overload.
+                for (var i = countParameters.Count - 1; i >= 0; i--)
+                {
+                    Parameter parameterToRemove = countParameters[i];
+                    newParameters.Remove(parameterToRemove);
+                }
+
+                // Get the new name for the overload.
+                int vectorStartIndexInName = vIndex - 1;
+                if (char.IsDigit(name[vIndex - 2]))
+                {
+                    vectorStartIndexInName -= 1;
+                }
```

Comment:
Why do we need this check?

Reply:
Because otherwise methods with a type between the number and the 'v' will fail. Like glUniform4fv

Reply:
Is there any guarantees that if the preceding character is a number that it's part of the vector "definition" and not just part of the function itself?

Reply:
No there is not. But the whole vector overloading probably needs a bit of work tbh. But i think we need to rely on the name somewhat for vector overloading.